devito/ir/equations/equation.py

```diff
@@ -206,6 +221,9 @@ def __new__(cls, *args, **kwargs):
         expr._reads, expr._writes = detect_io(expr)
         expr._implicit_dims = input_expr.implicit_dims
         expr._operation = Operation.detect(input_expr)
+        expr._target = input_expr.target if hasattr(input_expr, 'target') else None
```

Comment:
if you're using `hasattr` there's a 95% probability there's a much better way of doing this

first of all ask yourself why you're using `hasattr`

do we want all LoweredEq to have these new properties?
perhaps we want a new kind of LoweredEq?

I think this is worth some thinking